### ISSUE TITLE
AdminList table sorting problem with MySQL v5.7

### ISSUE THREAD

**Issue by iknowfoobar** [Kunstmaan/KunstmaanBundlesCMS#854]
I recently upgraded my iMac to use mysql v5.7 through homebrew. I noticed that when I tried to use any of the table sorting I got an exception

```
General error: 3065 Expression #1 of ORDER BY clause is not in SELECT list
```
and referring to the column that I tried to sort by. This seems to happen for all modules that I have created using an AdminList.

Reverting back to v5.5 which I use on live servers fixes the problem.

**Comment by genjusz** [comment]
You can change `sql_mode` of Your MySQL server to `NO_ENGINE_SUBSTITUTION`, but it would be nice if Kunstmaan worked with 5.7 without changing default server configuration.

**Comment by przemekkot** [comment]
try overiding `getPagerfanta` in your `AdminListConfigurator` and change:

`$adapter = new DoctrineORMAdapter($this->getQuery());`

to this:

`$adapter = new DoctrineORMAdapter($this->getQuery(), false);`

It should look like this:
```php
public function getPagerfanta()
    {
        if (is_null($this->pagerfanta)) {
            $adapter = new DoctrineORMAdapter($this->getQuery(), false);
            $this->pagerfanta = new Pagerfanta($adapter);
            $this->pagerfanta->setNormalizeOutOfRangePages(true);
            $this->pagerfanta->setCurrentPage($this->getPage());
            $this->pagerfanta->setMaxPerPage($this->getLimit());
        }

        return $this->pagerfanta;
 }
```

**Comment by Devolicious** [comment]
@genjusz @iknowfoobar This seems do be an issue with the doctrine paginator. Mysql 5.7 requires the field(s) within the select statement that are being used by the order by clause. The default behaviour of the doctrine paginator is to add a clone of the original query to fetch an array of ids.

The problem: https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryOutputWalker.php#L276

I opened an issue with doctrine for it https://github.com/doctrine/doctrine2/issues/5622

@przemekkot It's true that changing the fetchJoinCollection to false makes it work again, but the paginator will also behave differently as you can see here https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Tools/Pagination/Paginator.php#L141
For most cases it probably will not have any effect and work just fine.

**Comment by iknowfoobar** [comment]
Is there any update on when Kunstmaan will be compatible with MySQL 5.7 out of the box?

I just tried again and get this error when viewing the Media Library:

```code
An exception has been thrown during the rendering of a template ("An exception occurred while executing 'SELECT n.id, n.parent_id AS parent, t.url, t.id AS nt_id,
IF(t.weight IS NULL, v.weight, t.weight) AS weight,
IF(t.title IS NULL, v.title, t.title) AS title,
IF(t.online IS NULL, 0, t.online) AS online,
n.hidden_from_nav AS hidden,
n.ref_entity_name AS ref_entity_name FROM kuma_nodes n LEFT JOIN kuma_node_translations t ON (t.node_id = n.id AND t.lang = :lang) LEFT JOIN kuma_node_translations v ON (v.node_id = n.id AND v.lang <> :lang) INNER JOIN (SELECT DISTINCT o.object_identifier as id FROM testwebsite.acl_object_identities as o
INNER JOIN testwebsite.acl_classes c ON c.id = o.class_id
LEFT JOIN testwebsite.acl_entries e ON (
e.class_id = o.class_id AND (e.object_identity_id = o.id
OR e.object_identity_id IS NULL)
)
LEFT JOIN testwebsite.acl_security_identities s ON (
s.id = e.security_identity_id
)
WHERE c.class_type = "Kunstmaan\\NodeBundle\\Entity\\Node"
AND (s.identifier = "IS_AUTHENTICATED_ANONYMOUSLY" OR s.identifier = "ROLE_SUPER_ADMIN" OR s.identifier = "ROLE_PERMISSIONMANAGER" OR s.identifier = "ROLE_ADMIN" OR s.identifier = "ROLE_ALLOWED_TO_SWITCH" OR s.identifier = "Kunstmaan\\AdminBundle\\Entity\\User-admin")
AND e.mask & 1 > 0) perms_ ON perms_.id = n.id WHERE n.deleted = 0 GROUP BY n.id ORDER BY t.weight ASC, t.title ASC' with params ["en"]:

SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #3 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'testwebsite.t.url' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by") in KunstmaanAdminBundle:Default:app_header.html.twig at line 35
```

It still looks like I have to change the sql_mode to get it to work, although I have never reliably got this working. I have always had to revert back to 5.6

**Comment by jockri** [comment]
Currently, the only way to fix this problem on mysql 5.7:
In your mysql config remove the ONLY_FULL_GROUP_BY from the sql_mode parameter

There is some work that needs to be done to let the bundles work on 5.7 with the standard configuration.
